sdk/core/src/services/syft/mod.rs

```diff
@@ -0,0 +1,368 @@
+        match Command::new("syft")
+            .arg("--output")
+            .arg(CYCLONEDX_JSON_FORMAT)
+            .arg("--catalogers")
+            .arg(cataloger)
+            .arg(self.source_path.as_str())
+            .output()
+        {
+            Ok(output) => Ok(output),
+            Err(err) => Err(Error::SyftError(format!("Error executing Syft: {}", err))),
+        }
+    }
+
+    /// Conventional Constructor
+    pub fn new(source_path: String) -> Self {
+        Service { source_path }
+    }
+
+    /// Invokes the syft CLI against the cloned repository to generate an SBOM.
+    pub(crate) fn execute(
+        &self,
+        full_name: String,
+        commit_hash: String,
+        cataloger: Option<String>,
+        sub_path: Option<String>,
+    ) -> Result<String, Error> {
+        let mut orig_dir = None::<PathBuf>;
+
+        if let Some(path) = sub_path.clone() {
+            let absolute_path = format!("{}/{}", self.source_path, path.as_str());
+            let file_path = Path::new(&absolute_path);
+
+            orig_dir = match env::current_dir() {
+                Ok(path_buf) => Some(path_buf),
+                Err(err) => {
+                    return Err(Error::SyftError(format!(
+                        "Unable to get current directory: {}",
+                        err
+                    )))
+                }
+            };
+
+            env::set_current_dir(file_path).map_err(|err| {
+                Error::SyftError(format!(
+                    "Unable to change directories to build target location: {}",
+                    err
+                ))
+            })?
+        }
+
+        let output = match cataloger.clone() {
+            Some(cataloger) => self.run_syft_with_cataloger(cataloger),
+            None => self.run_syft(),
+        }?;
+
+        // Handle error generated by syft.
+        if !&output.status.success() {
+            return match String::from_utf8(output.stderr) {
+                Ok(stderr) => Err(Error::SyftError(format!(
+                    "Error executing Syft: {}",
+                    stderr
+                ))),
+                Err(err) => Err(Error::SyftError(format!(
+                    "Error executing Syft and reading stderr: {}",
+                    err
+                ))),
+            };
+        }
+
+        if output.stdout.is_empty() {
+            return Err(Error::SyftError("syft generated empty SBOM".to_string()));
+        };
+
+        let output: String = String::from_utf8_lossy(&output.stdout).to_string();
+        let mut json: Value =
+            serde_json::from_str(&output).map_err(|e| Error::SyftError(e.to_string()))?;
+
+        // Access the nested "component" object
+        if let Some(metadata) = json.get_mut("metadata").and_then(Value::as_object_mut) {
+            if let Some(component) = metadata.get_mut("component").and_then(Value::as_object_mut) {
+                component.insert(
```

Comment:
In Syft?  Always none remember? We can't get Syft to generate a purl for any of the CMS repos because is can't tell what those values might be.  I experimented a lot with Syft trying to get it to produce one and I can't even find anything in the documentation that says Syft can do such a thing.